src/Swarm/Language/Syntax.hs

```diff
@@ -426,79 +433,131 @@ isUserFunc c = case constMeta $ constInfo c of
 -- matching gives us warning if we add more constants.
 constInfo :: Const -> ConstInfo
 constInfo c = case c of
-  Wait -> commandLow 0
-  Noop -> commandLow 0
-  Selfdestruct -> commandLow 0
-  Move -> commandLow 0
-  Turn -> commandLow 1
-  Grab -> commandLow 0
-  Harvest -> commandLow 0
-  Place -> commandLow 1
-  Give -> commandLow 2
-  Install -> commandLow 2
-  Make -> commandLow 1
-  Has -> commandLow 1
-  Count -> commandLow 1
-  Reprogram -> commandLow 2
-  Drill -> commandLow 1
-  Build -> commandLow 2
-  Salvage -> commandLow 0
-  Say -> commandLow 1
-  Log -> commandLow 1
-  View -> commandLow 1
-  Appear -> commandLow 1
-  Create -> commandLow 1
-  Whereami -> commandLow 0
-  Blocked -> commandLow 0
-  Scan -> commandLow 0
-  Upload -> commandLow 1
-  Ishere -> commandLow 1
-  Self -> functionLow 0
-  Parent -> functionLow 0
-  Base -> functionLow 0
-  Whoami -> commandLow 0
-  Setname -> commandLow 1
-  Random -> commandLow 1
-  Run -> commandLow 1
-  Return -> commandLow 1
-  Try -> commandLow 2
-  Undefined -> functionLow 0
-  Fail -> functionLow 1
-  If -> functionLow 3
-  Inl -> functionLow 1
-  Inr -> functionLow 1
-  Case -> functionLow 3
-  Fst -> functionLow 1
-  Snd -> functionLow 1
-  Force -> functionLow 1
-  Not -> functionLow 1
-  Neg -> unaryOp "-" 7 P
-  Add -> binaryOp "+" 6 L
-  And -> binaryOp "&&" 3 R
-  Or -> binaryOp "||" 2 R
-  Sub -> binaryOp "-" 6 L
-  Mul -> binaryOp "*" 7 L
-  Div -> binaryOp "/" 7 L
-  Exp -> binaryOp "^" 8 R
-  Eq -> binaryOp "==" 4 N
-  Neq -> binaryOp "!=" 4 N
-  Lt -> binaryOp "<" 4 N
-  Gt -> binaryOp ">" 4 N
-  Leq -> binaryOp "<=" 4 N
-  Geq -> binaryOp ">=" 4 N
-  Format -> functionLow 1
-  Concat -> binaryOp "++" 6 R
-  AppF -> binaryOp "$" 0 R
-  Teleport -> commandLow 2
-  As -> commandLow 2
-  RobotNamed -> commandLow 1
-  RobotNumbered -> commandLow 1
-  Knows -> commandLow 1
+  Wait -> commandLow 0 "Wait for a number of time steps."
+  Noop ->
+    commandLow 0 . doc "Do nothing." $
+      [ "This is different than 'Wait' in that it does not take up a time step."
+      , "It is useful for commands like if, which requires you to provide both branches."
+      , "Usually it is automatically inserted where needed, so you do not have to worry about it."
+      ]
+  Selfdestruct ->
+    commandLow 0 . doc "Self-destruct the robot." $
+      [ "Useful to not clutter the world."
+      , "This destroys the robots inventory, so consider 'salvage' as an alternative."
+      ]
+  Move -> commandLow 0 "Move forward one step."
+  Turn -> commandLow 1 "Turn in some direction."
+  Grab -> commandLow 0 "Grab an item from the current location."
+  Harvest ->
+    commandLow 0 . doc "Harvest an item from the current location." $
+      [ "Leaves behind a growing seed if the harvested item is growable."
+      , "Otherwise it works exactly like 'grab'"
```

Comment:
```suggestion
      , "Otherwise it works exactly like 'grab'".
```

Reply:
Hah, your autocorrect put the period after the quotation mark. :smile: